### DISCUSSION TITLE
What does --fix-dry-run supposed to do?

### DISCUSSION THREAD

**Discussion by gtamas** [eslint/eslint#15668] [General]
I mean, I know it fixes auto fixable errors without really saving to disk.
But should it list all other (non-fixable) issues? Or only the fixable ones?

We found that running eslint without the --fix-dry-run option sometimes lists more errors.
How is this supposed to work exactly?
This is Eslint 7.32.0.

**Comment by mdjermanovic** [comment]
The [`--fix-dry-run`](https://eslint.org/docs/user-guide/command-line-interface#--fix-dry-run) option is mostly for integrations via `stdin`.

When you use this option with the default formatter, you'll see the non-fixable issues that remain after the fixable ones are fixed. The results should be exactly the same as with the `--fix` option, just without saving the file.

(this is a bit simplified explanation; fixes might have conflicts with other fixes, in which case they won't be applied in the same iteration; ESLint runs max 10 iterations, so theoretically some fixable issues may also remain).

**Reply by gtamas** [reply]
OK thanks for the explanation.
